packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -5,23 +5,25 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import { TextDocumentSaveReason } from "vscode-languageserver-types";
+import { TextDocumentSaveReason, Position } from "vscode-languageserver-types";
+import { MonacoToProtocolConverter, ProtocolToMonacoConverter } from "monaco-languageclient";
+import { TextEditorDocument } from "@theia/editor/lib/browser";
 import { DisposableCollection, Disposable, Emitter, Event, Resource } from '@theia/core/lib/common';
 import ITextEditorModel = monaco.editor.ITextEditorModel;
 
 export {
     TextDocumentSaveReason
-}
+};
 
 export interface WillSaveModelEvent {
     readonly model: monaco.editor.IModel;
     readonly reason: TextDocumentSaveReason;
     waitUntil(thenable: Thenable<monaco.editor.IIdentifiedSingleEditOperation[]>): void;
 }
 
-export class MonacoEditorModel implements ITextEditorModel {
+export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
 
-    autoSave: boolean = true;
+    autoSave: 'on' | 'off' = 'on';
```

Comment:
I would like to move autosave functionality to `@theia/core`, i.e. it will work based `Saveable` and applied to all widgets, not only to editors. Objections?

Reply:
Will it be mandatory for all editing widgets to support both modes?

Reply:
I thought it would be supported automatically and configured with one common preference. But thinking again it is probably should be configurable per a widget type.

Reply:
Ah, you simply want trigger save when `onDirtyChanged` signals?

Reply:
We've discussed that it will require moving prefereces-api to core. I would like to make it as the separate PR.